### PR TITLE
Fixed sifter mesh buy option for creative mode.

### PR DESCRIPTION
**Pull request by GeldornOfDragonhill**
Also fixed the color setting of the labels.

Fixes the Problem in #4917 for the 1.15.2 version.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/network/messages/server/colony/building/sifter/SifterSettingsMessage.java [hunk 1]
```diff
@@ -105,7 +105,9 @@ protected void onExecute(
             final PlayerEntity player = ctxIn.getSender();
             if (player == null) return;
 
-            InventoryUtils.reduceStackInItemHandler(new InvWrapper(player.inventory), mesh);
+            if(!player.isCreative()) {
+                InventoryUtils.reduceStackInItemHandler(new InvWrapper(player.inventory), mesh);
```
Comment (Asherslab) [inline src/main/java/com/minecolonies/coremod/network/messages/server/colony/building/sifter/SifterSettingsMessage.java]: This is correct, however i just noticed that we don't check that the player (if in survival) has the item in their inventory on server side. As such this is exploitable (free upgrades) by a client using a bad version of the mod.

Can you please add a check up the top of this method to return if the player doesn't have the required item in their inventory (or ignore if in creative)

Thanks

**Review by Asherslab — CHANGES_REQUESTED** [review]

**Comment by LDTTeam-Buildsystem** [comment]
This PR has been automatically built and hosted by our build system, during this process there was a crash, error, or otherwise, please check on this at: https://teamcity.minecolonies.com/viewLog.html?buildId=39574&buildTypeId=LetSDevTogether_Minecolonies_PullRequests_2_DeployTestServer

**Comment by GeldornOfDragonhill** [comment]
Added the server side check, should the player/log be notified of the error?
A quick check with the warehouse revealed a similar problem.

**Review by Asherslab — DISMISSED** [review]

**Comment by LDTTeam-Buildsystem** [comment]
This PR has been automatically built and hosted by our build system, during this process there was a crash, error, or otherwise, please check on this at: https://teamcity.minecolonies.com/viewLog.html?buildId=39577&buildTypeId=LetSDevTogether_Minecolonies_PullRequests_2_DeployTestServer

File: src/main/java/com/minecolonies/coremod/network/messages/server/colony/building/sifter/SifterSettingsMessage.java [hunk 2]
```diff
@@ -98,14 +99,26 @@ protected void onExecute(
...
         }
-        building.setup(new ItemStorage(block), new ItemStorage(mesh), qty);
 
         if (buy)
         {
             final PlayerEntity player = ctxIn.getSender();
             if (player == null) return;
 
-            InventoryUtils.reduceStackInItemHandler(new InvWrapper(player.inventory), mesh);
+            if(!player.isCreative()) {
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/network/messages/server/colony/building/sifter/SifterSettingsMessage.java]: just need to fix the { formatting here

**Comment by GeldornOfDragonhill** [comment]
Coding style fixes applied.

**Comment by LDTTeam-Buildsystem** [comment]
This PR has been automatically built and hosted by our build system, during this process there was a crash, error, or otherwise, please check on this at: https://teamcity.minecolonies.com/viewLog.html?buildId=39633&buildTypeId=LetSDevTogether_Minecolonies_PullRequests_2_DeployTestServer

**Pull request merged**